client/app/templates/templates.service.js

```diff
@@ -73,4 +74,17 @@ export function TemplatesServiceFactory(CollectionsApi) {
 
     return queryFilters;
   }
+  function createTemplate(template) {
+    return new Promise(function (resolve, reject) {
+      CollectionsApi.post(collection, null, {}, template).then(createSuccess, createFailure);
+
+      function createSuccess(response) {
```

Comment:
also handling success and failure should happen in the component, not in the service